avocado/utils/memory.py

```diff
@@ -27,7 +27,95 @@
 from . import genio
 
 
-# Returns total memory in kb
+class MemoryError(Exception):
+    """
+    called when memory operations fails
+    """
+    pass
+
+
+def _check_memory_state(block):
+    """
+    Check the given memory block is online or offline
+
+    :param block: memory block id.
+    :type string: like 198
+    :return: 'True' if online or 'False' if offline
+    :rtype: bool
+    """
+    while True:
```

Comment:
This is too aggressive. Not sure if the `state` can stay for a long time in 'going-online' or 'going-offline', but the fact is that you should wait nicely, with a timeout and steps that will avoid killing the cpu in charge of the wait process.
We have a `wait` lib in our utils, it might be useful here. You could have something like:

```python
from avocado.utils import wait
...

def _check_memory_state(block):
    def _is_online():
        # Use `with open()` here. If block is online, return True. Else, return False.
        # No need to check 'going-online'/'going-offline' here.
        # No need to handle exception, IMO. Let users receive the exceptions
        # and handle them, if they want to.

    return wait.wait_for(_is_online, timeout=10, step=1) or False
```